ts/src/vertex.ts

```diff
@@ -0,0 +1,2796 @@
+        //       "digest": {order digest}
+        //     },
+        //     "request_type": "execute_place_order"
+        //     "id": 100
+        // }
+        //
+        const data = this.safeDict (response, 'data', {});
+        return this.safeOrder ({
+            'id': this.safeString (data, 'digest'),
+        });
+    }
+
+    parseOrderStatus (status) {
+        if (status !== undefined) {
+            const statuses = {
+                'pending': 'open',
+            };
+            if (typeof status === 'string') {
+                return this.safeString (statuses, status, status);
+            }
+            const statusCancelled = this.safeDict (status, 'cancelled');
+            if (statusCancelled !== undefined) {
+                return 'canceled';
+            }
+            const statusTriggered = this.safeDict (status, 'triggered', {});
+            const triggeredStatus = this.safeString (statusTriggered, 'status', 'failure');
+            if (triggeredStatus === 'success') {
+                return 'closed';
+            }
+            return 'canceled';
+        }
+        return status;
+    }
+
+    parseOrder (order, market: Market = undefined): Order {
+        //
+        // {
+        //     "product_id": 1,
+        //     "sender": "0x7a5ec2748e9065794491a8d29dcf3f9edb8d7c43000000000000000000000000",
+        //     "price_x18": "1000000000000000000",
+        //     "amount": "1000000000000000000",
+        //     "expiration": "2000000000",
+        //     "nonce": "1",
+        //     "unfilled_amount": "1000000000000000000",
+        //     "digest": "0x0000000000000000000000000000000000000000000000000000000000000000",
+        //     "placed_at": 1681951347,
+        //     "order_type": "ioc"
+        // }
+        // stop order
+        // {
+        //     "order": {
+        //       "order": {
+        //         "sender": "0x7a5ec2748e9065794491a8d29dcf3f9edb8d7c43000000000000000000000000",
+        //         "priceX18": "1000000000000000000",
+        //         "amount": "1000000000000000000",
+        //         "expiration": "2000000000",
+        //         "nonce": "1",
+        //       },
+        //       "signature": "0x...",
+        //       "product_id": 1,
+        //       "spot_leverage": true,
+        //       "trigger": {
+        //         "price_above": "1000000000000000000"
+        //       },
+        //       "digest": "0x..."
+        //     },
+        //     "status": "pending",
+        //     "updated_at": 1688768157050
+        // }
+        //
+        let marketId = this.safeString (order, 'product_id');
+        let timestamp = this.safeTimestamp (order, 'placed_at');
+        let amount = this.safeString (order, 'amount');
+        let price = this.safeString (order, 'price_X18');
+        const remaining = this.safeString (order, 'unfilled_amount');
+        let triggerPriceNum = undefined;
+        const status = this.safeValue (order, 'status');
```

Comment:
why safeValue here?

Reply:
@carlosmiei the status would be string and object.

Reply:
ok ok